### DISCUSSION TITLE
Does supabase-js and its createClient benefits from Next 13's memoized requests?

### DISCUSSION THREAD

**Discussion by onlyphantom** [supabase/supabase#17669] [Questions] [answered]
Next 13 has a feature called [Request Memoization](https://nextjs.org/docs/app/building-your-application/caching#request-memoization), which:

> React extends the [fetch API](https://nextjs.org/docs/app/building-your-application/caching#fetch) to automatically memoize requests that have the same URL and options. This means you can call a fetch function for the same data in multiple places in a React component tree while only executing it once.

![image](https://github.com/supabase/supabase/assets/16984453/2b8ead2e-6562-4b22-8ff9-1fb97549abe8)

> For example, if you need to use the same data across a route (e.g. in a Layout, Page, and multiple components), you do not have to fetch data at the top of the tree then forward props between components. Instead, you can fetch data in the components that need it without worrying about the performance implications of making multiple requests across the network for the same data.

I wonder if using `supabase-js`, particularly with the `createClient` wrapper also automatically inherit the benefit of JavaScript's own fetch API. If so, making a db trip from multiple pages, assuming for the same table, would in turn use the memoized value (so a grand sum of just ONE call instead of several calls). Obviously, if that is the case, it's a big difference in build time and costs as well for a non-trivial project.

Thanks!

**Comment by RyelBanfield** [comment]
I'm wondering this as well. Figured it out yet?

**Reply by RyelBanfield** [reply, marked as answer]
I tested this by making a call to getUser in both my Nav bar and my Page.

In development, I saw two calls to /auth in the supabase logs.. but when I built the app and ran it in prod, I only saw one call to /auth.
